tests/compile/test_noop_elimination.py

```diff
@@ -25,10 +25,13 @@ def test_noop_elimination(dtype, num_tokens, hidden_size, buffer_size):
     class Model(torch.nn.Module):
         def __init__(self) -> None:
             super().__init__()
-            self.pos_embed = torch.empty(buffer_size, hidden_size, dtype=dtype)
+            # Avoid using empty, since on rocm torch.empty
+            # does not initialize the memory.
+            self.pos_embed = torch.randn(buffer_size, hidden_size, dtype=dtype)
 
         def forward(self, x):
-            x += self.pos_embed[: x.shape[0]]
+            # Avoid += to prevent inplace addition.
+            x = x + self.pos_embed[: x.shape[0]]
```

Comment:
Why is this change necessary?

Reply:
'''
x[0][0]: tensor(0.6719, device='cuda:0')
embed[0][0]: tensor(-0.2964, device='cuda:0')
result[0][0][0]: tensor(0.3755, device='cuda:0')
result2[0][0][0]: tensor(0.0791, device='cuda:0')
'''
result = x + embed
result2 = x + embed + embed

`+=` is doing inplace addition will cause all test cases fail. On NV side, because `torch.empty` always initializes memory to zero, so we didn't see the failure before.

Reply:
But since on Rocm, we can not use `torch.empty()` since it does not initialize memory, so the value could be nan, causing test failure.